type-check.ts

```diff
@@ -212,7 +212,26 @@ export function tcStmt(env : GlobalTypeEnv, locals : LocalTypeEnv, stmt : Stmt<n
         throw new TypeCheckError("Condition Expression Must be a bool");
       return {a: NONE, tag:stmt.tag, cond: tCond, body: tBody};
     case "pass":
+    case "break":
+    case "continue":
       return {a: NONE, tag: stmt.tag};
+    case "for":
+      var tIterator = tcExpr(env, locals, stmt.iterator)
+      if(tIterator.tag!="id")
+        throw new TypeCheckError("iterator must be an id");
+      var tValObject = tcExpr(env, locals, stmt.values)
+      if (tValObject.a.tag !== "class") 
+        throw new TypeCheckError("values require an object");
```

Comment:
Nice. This is a good place to generalize a bit when we get to lists and strings, etc.

In general I appreciate how straightforward this type-checking is. Easy for me to see how it will be extended/made more flexible in the future.